### DISCUSSION TITLE
YOLOv8-obb

### DISCUSSION THREAD

**Discussion by hellotenet** [ultralytics/ultralytics#11619] [General]
Hello, can I use parts of YOLOv8-obb to compute AP and mAP for rotated bounding box object detection? What I mean is that I will input the information annotated by the machine (p) and the ground truth (gt).

**Comment by pderrenger** [comment]
Hello! Yes, you can indeed use YOLOv8-obb for computing AP and mAP for rotated bounding boxes. You just need to run the model prediction and compare these predictions with your ground truth. Here’s a simple example of how you might set this up:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('path/to/yolov8-obb.pt')

# Perform predictions
results = model('path/to/your/image_or_video')

# Process the results for computing AP and mAP
# An example of accessing predicted boxes
predicted_boxes = results.pred[0]

# Here you would also load and process your ground truth data, and compute AP/mAP
```

Make sure your predictions and ground truth data are matched correctly for accurate AP and mAP calculations. Happy coding! 🚀

**Reply by hellotenet** [reply]
Sorry if I didn't express my meaning clearly. I used YOLOv8-obb to obtain prediction results, and then performed certain post-processing on these results without altering the data format. Can I still utilize some functionalities of YOLOv8-obb, inputting my own post-processing results (p) and the ground truth annotations (gt), to calculate AP and mAP.

**Reply by pderrenger** [reply]
Absolutely! You can indeed calculate AP and mAP using your post-processed results from YOLOv8-obb. Just ensure that the post-processed results (`p`) maintain the correct format expected by the evaluation functions, and then proceed to compare them with the ground truth (`gt`). Here’s a basic setup using Python:

```python
from ultralytics import YOLO, evaluate_ap

# Load model
model = YOLO('path/to/yolov8-obb.pt')

# Assume 'results' contains your post-processed predictions
# your_processed_results should be in the format ready for evaluation
your_processed_results = process_results(results)

# Load your ground truth in a compatible format
ground_truth = load_ground_truth('path/to/gt.data')

# Calculate AP and mAP
ap, map = evaluate_ap(your_processed_results, ground_truth)
```

Just replace `process_results` and `load_ground_truth` with your actual data processing functions. Happy analyzing! 🚀

**Reply by hellotenet** [reply]
from ultralytics import YOLO, evaluate_ap

results_path = 'labels'
results = process_results(results_path)

ground_truth = load_ground_truth('labels')

ap, map = evaluate_ap(results, ground_truth)

ImportError: cannot import name 'evaluate_ap' from 'ultralytics' (C:\Users\Lenovo.conda\envs\yolo\Lib\site-packages\ultralytics_init_.py)

thanks. can you help me solve this problem

**Reply by pderrenger** [reply]
@hellotenet hello! It looks like the `evaluate_ap` function is not available directly from the `ultralytics` module. Instead of importing `evaluate_ap`, you typically handle AP and mAP calculations using the evaluation functionality integrated within the training and validation processes provided by YOLOv8.

Here's a simple way to evaluate your model using the built-in validation:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8-obb.pt')

# Perform validation which will automatically compute AP and mAP across your dataset
results = model.val(data='your_validation_data.yaml')
print("mAP:", results.map)
```

Ensure that you have your validation dataset configured properly in a YAML file. Happy coding! 🙂

**Reply by hellotenet** [reply]
Excuse me, when using a self-trained YOLO-OBB for prediction, is it possible to return a TXT document in DOTA format for the labels?

**Reply by pderrenger** [reply]
@hellotenet hi there! Currently, YOLOv8 does not directly support exporting predictions in DOTA format out-of-the-box. However, you can easily script a converter to reshape the output into DOTA format. Here’s a simple example in Python:

```python
def convert_to_dota_format(results):
    # Assume results contain boxes in xywh format
    with open('output.txt', 'w') as file:
        for box in results.boxes.xywh:
            x, y, w, h = box
            # Convert xywh to DOTA format and write to file
            file.write(f'{x} {y} {x+w} {y+h}\n')  # Simplified, adjust as per DOTA requirements

# Your code to get predictions
# results = your_model_predict_function(...)
convert_to_dota_format(results)
```
Just adjust the format inside the function as needed to match specific DOTA requirements. Let me know if there's anything else you need! 😊
